spec/api/api_put_spec.rb

```diff
@@ -0,0 +1,167 @@
+require_relative '../spec_helper'
+require_relative 'api_client.rb'
+require 'securerandom'
+
+RSpec.describe 'PUT method checking to the any data in the body' do
+	api_cl = ApiClient.new
+	body = api_cl.generate_random_body
+	
+	before(:all) do
+		api_cl.create_user(body)
+	end
+	
+	# Positive Functional Testing
+	
+	it 'verifies that username can be edited' do
+		body[:username] = 'KingOfTheNorth'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that first name can be edited' do
+		body[:firstName] = 'Robb'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that last name can be edited' do
+		body[:lastName] = 'Stark'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that email can be edited' do
+		body['email'] = 'robb-stark@gameofthrones.com'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that password can be edited' do
+		body[:password] = 'rob-stark@gameofthrones.com'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that phone number can be edited' do
+		body[:phone] = rand(100000).to_s
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	# Negative functional testing
+	# Username
+	it 'verifies that username can not be less than 2 symbols' do
+		body[:username] = 'u'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
```

Comment:
No precision in this check at all, u expect anything but 200. Doesn't tell us what's happening.
In case there is error raised, You can use https://relishapp.com/rspec/rspec-expectations/v/2-11/docs/built-in-matchers/raise-error-matcher. In case there is error code only, check for it.

Reply:
Applies to all such checks